src/audio/pipeline/pipeline-schedule.c

```diff
@@ -180,10 +181,10 @@ void pipeline_schedule_triggered(struct pipeline_walk_context *ctx,
 			p = container_of(tlist, struct pipeline, list);
 			if (pipeline_is_timer_driven(p)) {
 				/* Use the first of connected pipelines to trigger */
-				if (cmd >= 0) {
+				if (cmd >= 0 && !first_pipe) {
```

Comment:
Why the cmd is being checked for >= 0 here? control won't come here if the switch case doesn't match

Reply:
humm, yes, maybe we only need the !first _pipe. @lyakh would you concur?

Reply:
I actually don't find this "comment by cppcheck" very useful... yes, `cmd` is used as a local variable here, it absolutely isn't expected to have any effect outside of this function. But ok, using a separate variable makes this more explicit. But otherwise - yes, after this change the `cmd >= 0` test can be removed there. But `first_pipe` also seems inverted to me: it is `false` *while* the loop is handling the *first pipeline* and it becomes true for the second and any further pipelines. Shouldn't this be inverted?

Reply:
yes, I still don't know my left from my right... will fix, thanks @lyakh

Reply:
fix in https://github.com/thesofproject/sof/pull/4832